File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        dct = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
        maxVowels = sum(1 for i in range(k) if s[i] in dct)
        currVowels = maxVowels

        for right in range(k, len(s)):
            if s[right] in dct:
                currVowels += 1
            if s[right - k] in dct:
                currVowels -= 1
            maxVowels = max(maxVowels, currVowels)
        return maxVowels

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        Set<Character>  hs = new HashSet<>();
        hs.add('a'); hs.add('e'); hs.add('i'); hs.add('o'); hs.add('u');
        int maximum = 0, curr = 0;
        for(int right = 0; right< k;right++){
            if (hs.contains(s.charAt(right))){
                curr++;
            }
        }
        maximum = curr;
        for(int right = k; right < s.length(); right++){
            if (hs.contains(s.charAt(right))){
                curr++;
            }
            if (hs.contains(s.charAt(right - k))){
                curr--;
            }
            maximum = Math.max(maximum, curr);
        }
        return maximum;
        
    }
}